content/projects/_index.md

```diff
@@ -4,9 +4,75 @@ description: "This page provides the details of SODA Projects"
+    The vision is to build a unified intelligent and scalable infrastructure management framework for resource management (config, add, remove, update), alert management, and performance metrics management with standardized APIs across any storage and application platforms, with connectors for data analysis and prediction.
+
+- ### SODA API
+    Provides the standardization for Data / Storage Management APIs. Currently we support block and file APIs for key features of data management (provisioning, migration, fileshare, etc). Working to add the storage management APIs.
+
+    This is the key external interface to platforms, which can do a seamless integration with heterogeneous storage backends. 
+
+    Users can develop SODA North-Bound Plugins (SODA NBP) under SODA NBP project to connect any platform or application solutions to SODA API from north for all storage/data requirements.
+
+    We envision this to be the reference implementation of SODA Data Standard API Specification, which we plan to work with our industry partners and standards bodies. At that stage, this layer will upgraded to support Block, File and Object APIs across the Edge, Core and Cloud.
+
+- ### SODA Controller
+    Provides metadata management, control/configurations, scheduler and all such bookkeeping features and utilities. In the API flow from SODA API to SODA Dock, controller plays a critical role for all the API flow management and tracking to handle all the state machine and metadata management requirements. This will be a layer to keep addons to new facilities or utilities for the soda platform.
+
+    Currently we think, this layer can be optional or similar modules can be used from users(if they have such software components). However, the users needs to do certain integration with SODA API and SODA Dock for their controller modules.
+
+    Controller interfaces with SODA API and SODA Dock. Currently uses grpc and we are working to refine this interface (REST/Msg etc)
+
+- ### SODA Dock
+    SODA Dock is a docking station for heterogeneous storage backends! This is where all the different storage vendors drivers for various storage backend models get attached.
+
+    The goal is to make most of the protocols and backends supported as close as ‘plug n play’. Currently, each storage backend needs a thin, easy to develop SODA Driver Plugin to connect the storage backend to the Dock. SODA Driver Plugin and Storage vendor driver together it can be called SODA Driver for xxx vendor yy model storage.SODA Driver can support one or more or multiple classes of storage backends.
+
+    To connect more and more cloud-native storages seamlessly with the SODA Platform,there is an experimental feature for CSI PlugNPlay is getting developed. This will immediately enable any CSI driver to work with SODA using a single SODA CSI plugin for the application platform (like Kubernetes), in turn, support all the storage vendors supporting CSI.
```

Comment:
_there is an experimental feature for CSI PlugNPlay is getting developed._
->
_an experimental feature, CSI PlugNPlay is being developed._